src/charm.py

```diff
@@ -27,6 +27,10 @@
 class MySQLRouterOperatorCharm(ops.CharmBase):
     """Operator charm for MySQL Router"""
 
+    # Track if unit has previously handled a mysql_router_pebble_ready event
+    # (to detect pod restart)
+    _stored = ops.framework.StoredState()
```

Comment:
is there any reason why we're using stored state instead of a peer relation?

Reply:
I remember a caution against using StoredState in charms, and how it is generally better to use peer relation databags instead. Please refer to [the following article](https://discourse.charmhub.io/t/storedstate-uses-limitations/5740) for the caution

In particular:
> Let’s say the charm is running on Kubernetes, and the container it is running in gets destroyed and recreated. This might happen due to events outside of an operator’s control – perhaps the underlying Kubernetes service rescheduled the pod, for example. In this scenario the StoredState will go away, and the flags will be reset.

Although, there is seemingly conflicting messaging [in this article](https://discourse.charmhub.io/t/keeping-state-in-juju-controllers-in-operator-framework/3303)
> In general, there should be very few other changes that a charm needs to make. All of the StoredState() objects in the charm (including the one the Framework itself uses) will now use this for their storage. Which means that state will persist to restarted K8s pods, etc.

Additionally, I think that using the local unit databag in a peer relation is an equivalent way of storing state for the unit (the peer relation need not only be used to share data amongst the units)

If we conclude to keep using StoredState, we should probably consider a unit being scaled down and then scaled back up - would the stored state be cleared appropriately (as the state is stored in the juju controller)?